### PR TITLE
Tiny optimzation: Removed unnecesary intermediary list creation

### PR DESCRIPTION
**Pull request by CreamyCookie**
Check-list
----------

 * [ ] There is a linked issue discussing the motivations for this feature or bugfix
 * [x] Unit tests have been added where possible
 * [x] I've added / updated documentation for any user-facing features.
 * [x] Performance seems to be comparable or **better** than current `master`

What this change is adding / fixing
-----------------------------------
It's removing a single intermediary list creation and thus also avoids iterating over all songs twice!

### CODE REVIEW

**Comment by CreamyCookie** [comment]
@declension Is there anything we can do about `tests.test_util_copool.Tcopool testMethod=test_pause_restart_pause`?

File: quodlibet/util/songwrapper.py [hunk 1]
```diff
@@ -128,9 +128,10 @@ def check_wrapper_changed(library, parent, songs):
 
 
 def background_check_wrapper_changed(library, songs):
-    need_write = [s for s in songs if s._needs_write]
+    for song in songs:
+        if not s._needs_write:
```
Comment (declension) [inline quodlibet/util/songwrapper.py:132]:
```suggestion
        if not song._needs_write:
```

Comment (CreamyCookie) [reply quodlibet/util/songwrapper.py:132]: oh my.. why did I miss this? thank you!

**Comment by declension** [comment]
> @declension Is there anything we can do about `tests.test_util_copool.Tcopool testMethod=test_pause_restart_pause`?

It's pretty annoying. Added https://github.com/quodlibet/quodlibet/issues/3653

**Comment by CreamyCookie** [comment]
@declension Thanks for the review and creating #3653!

> It's pretty annoying

Agreed. Requires you to look at the logs all the time, even if everything's alright.. (as in the last change by you)

**Pull request merged**